startreedata/thirdeye
# [pinot] add default query option support

Add support for default query options. 
Default query options are injected by the ThirdEyeDataSource. 
They can be overriden by consumers (eg cohort reco, alert config level) by passing custom options in the `DataSourceRequest` when using `ThirdEyeDataSource#fetchDataTable`.

`defaultQueryOptions` is a Map<String,String>, the ThirdEyeDataSource is responsible for injecting them in the good format. 

To test the feature: 
add 
```
"defaultQueryOptions": {
    "timeoutMs": "1"
  },
```
to your DataSource

--> go on an alert, whatch your queries timeout.


See pinot query options here: 
https://docs.pinot.apache.org/users/user-guide-query/query-options 

NEXT: 
- [ ] consider having a generic parameter for the timeOut? to make usage by clients simpler and not leak pinot knowledge?
- [ ] when this is merged - update StarTreeThirdeye
- [ ] add better default/custom value for cohort recommender
- [ ] add per/alert custom options
- [ ] change some defaults in the deployer/k8s side?
